src/test/kotlin/com/pubnub/api/legacy/endpoints/pubsub/PublishTest.kt

```diff
@@ -562,4 +534,84 @@ class PublishTest : BaseTest() {
         assertEquals("0", requests[0].queryParameter("store").firstValue())
         assertFalse(requests[0].queryParameter("ttl").isPresent)
     }
+
+    @Test
+    fun testSpaceIdQueryParamIsPassedInPublish() {
+        val spaceIdValue = "thisIsSpaceId"
+
+        stubFor(
+            get(urlPathEqualTo("/publish/myPublishKey/mySubscribeKey/0/coolChannel/0/%22hi%22"))
```

Comment:
What about extracting "coolChannel" to variable. Then the variable can be used here and in publish call:
channel = "coolChannel",
?

Reply:
Sure. Why not